xds/src/main/java/io/grpc/xds/client/XdsClientImpl.java

```diff
@@ -263,12 +234,10 @@ public void run() {
         ImmutableMap.Builder<XdsResourceType<?>, Map<String, ResourceMetadata>> metadataSnapshot =
             ImmutableMap.builder();
         for (XdsResourceType<?> resourceType: resourceSubscribers.keySet()) {
-          ImmutableMap.Builder<String, ResourceMetadata> metadataMap = ImmutableMap.builder();
```

Comment:
Why was this changed? It looks like previously we returned an immutable map and now we don't. Is something depending on it being mutable now?